documentation/internal/Element-Type-Specs.md

```diff
@@ -0,0 +1,129 @@
+_Ongoing authoritative reference of Template Question/Element types, including input parameters and response type (shape). Please ensure this document matches the current implementation at all times._
+
+## Template Element fields
+
+- **id, section_id** : database references
+- **code**: `string` -- unique (per template) identifier
+- **index**: `integer` -- element ordering sequence (starting at `0`)
+- **title**: `string` -- describes the element, shown in template builder (probably)
+- **category**: `enum` -- either "Information" or "Question"
+- **visibility_condition**: `JSON` -- dynamic query determining whether element appears to the user
+  - default: `{"value": true}` (i.e. it always appears)
+- **element_type_plugin_code**: `string` -- code of the question/element plugin that presents this element to the UI
+- **is_required**: `JSON` -- dynamic query determining whether question is required to be answered
+  - default: `{"value": true}`
+- **is_editable**: `JSON` -- dynamic query determining whether can be edited (Would only be false in rare circumstances)
+  - default: `{"value": true}`
+- **parameters**: `JSON` -- the parameters specific to each question/element type. See individual plugins below for parameter breakdown
+- **validation**: `JSON` -- a dynamic expression for checking if the user's response is a valid input.
+  - default: `{"value": true}`
+- **validation_message**: `string` -- the message that shows in the UI when validation fails.
+
+## Question/Element types
+
+**Note**: all parameter fields can also have a dynamic query object instead of a primitive. The `evaluateExpression` function will return literal strings (or numbers, booleans) as is. The types described for the parameters below are the type that is expected to be _returned_ from a query expression.
+
+### Short Text Input _(not yet implemented -- highest priority)_
+
+- **type/code**: `shortText`
+- **category**: `Question`
+
+_Free-form, single-line text input element_
+
+#### Input parameters (in the `parameters` JSON)
+
+- **label**: `string` -- Text that shows in the HTML "label" attribute of the form element (suggestion: make this interpret Markdown)
```

Comment:
> **label** - [...] suggestion: make this interpret Markdown

I have added a issue regarding this suggestion #86

Reply:
Thanks.